### PR TITLE
Feature: pass raw bytes from native, parse using dart-opendroneid

### PR DESCRIPTION
**Pull request by matejglejtek**
Remove parsing logic from native parts of the app, remove all models and enums representing messages and values in them. Rename `MessagePack` to `MessageContainer` to avoid clash with dart-opendroneid's `MessagePack`.

Add `ODIDPayload` class that is sent from native to dart. It contains raw data and metadata such as source and mac address. In dart code, parse the data using `dart-opendroneid` into messages and save them to `MessageContainer`.

To test this,~~edit path in `pubspec.yaml` to point to your cloned `dart-opendroneid` repository~~.
Clone **DroneScanner** branch [use-updated-odid](https://github.com/dronetag/drone-scanner/tree/feature/use-updated-odid-lib) and run it with this version of `flutter-opendroneid`.

TODO

- [x] test all possible sources - bt5, wi-fi - I will test wi-fi ODID later this week with help of David
- [x] make sure duplicate messages filtered out
- [x] refactor kotlin scanner classes
- [x] test `MessagePacks`, make sure they transform to `MessageContainer` correctly - error found in Dart-opendroneid `MessagePack` parsing - [PR1](https://github.com/dronetag/dart-opendroneid/pull/1).
- [x] wait for release of dart-opendroneid and change dependency
- [x] update documentation

### CODE REVIEW

**Comment by matejglejtek** [comment]
**Update:**

I tried scanning for data from all sources including simulated Wi-FI scan using ESP and achieved the same results as with the original version. Parsing of all message types including `MessagePack` works. I found few issues with dart-odid and immediately solved them ([PR1](https://github.com/dronetag/dart-opendroneid/pull/1))

Waiting for release of dart-odid.

**Comment by marianhlavac** [comment]
Pls don't forget to assign reviewers, this was the PRs gets easily missed.

**Comment by marianhlavac** [comment]
dart_opendroneid is released: https://pub.dev/packages/dart_opendroneid/versions/0.1.0

**Comment by matejglejtek** [comment]
I used the released version of dart_opendroneid. But since [PR1](https://github.com/dronetag/dart-opendroneid/pull/1) was not resolved yet, `MessagePacks` wont be parsed correctly and BT data wont be parsed on some phones.

File: lib/models/message_container.dart
Comment (albertmoravec) [inline lib/models/message_container.dart:6]: Please add a short description of this class' purpose (especially why it contains all the message types).

I actually think that this is how `MessagePack` should've been implemented in `dart-opendroneid` (each message type appearing once or not at all). Will check with @optical-o .

File: lib/models/message_container.dart
Comment (albertmoravec) [inline lib/models/message_container.dart:146]: This one might be a getter instead? The checks below might be as well if you really want.

File: android/src/main/kotlin/cz/dronetag/flutter_opendroneid/scanner/ODIDScanner.kt
Comment (albertmoravec) [inline android/src/main/kotlin/cz/dronetag/flutter_opendroneid/scanner/ODIDScanner.kt:20]: This might go to the `WifiNaNScanner` companion object instead? Same for BT if there's something.

**Review by albertmoravec — APPROVED** [review]
I gave some comments on very minor issues, but I'll let you decide.

Also do you think that maybe we could move some of the conversion and comparison code from `compare_extensions.dart` and `conversions.dart` to `dart-opendroneid`?

File: lib/models/message_container.dart
Comment (matejglejtek) [reply lib/models/message_container.dart:146]: yes i will use getters

**Comment by matejglejtek** [comment]
I used the getters in `MessageContainer`, renamed method `getPayload` to `buildPayload`. Maybe we could create new task to resolve situation with `MessagePack` vs `MessageContainer` and moving conversions to **dart-opendroneid**?

**Comment by marianhlavac** [comment]
> Maybe we could create new task to resolve situation with MessagePack vs MessageContainer and moving conversions to dart-opendroneid?

Yeah, this will be ultimately resolved by also joining the structures that are used by the upcoming 'DRI Receiver' product. We don't have to resolve it right now.

**Review by marianhlavac — APPROVED** [review]
**Looks great!**

Tested on my iPhone X, data seems also ok, but more thorough testing will be required in the future.

We will follow up with the structures for the metadata (the new container class you've created). There is an opportunity to use unified structures, as 'DRI Receiver' and potentially mobile app could use them too.

**Pull request merged**
